### PR TITLE
[VITS] Fix nightly tests

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Fixes the tokenizer integration test and multi-GPU test that failed on the nightly run: https://github.com/huggingface/transformers/actions/runs/6068405445/job/16461410319

The tokenizer fix is trivial (needed to update the commit ID)!

The multi-GPU test was failing because the output sequence length for VITS is a function of the model **inputs**, rather than being a function of the input **sequence lengths** only.

Let's say we have 2 GPUs over which we want to run DP:
* GPU 1 outputs a sequence length of `N`, which is computed based on the input in the first element of the batch `x`
* GPU 2 outputs a sequence length of `M`, which is computed based on the input in the second element of the batch `y`

=> there is nothing to enforce that `N = M`, since the VITS output sequence length is a function of the inputs. Thus, we cannot concatenate the inputs after running the forward pass, since they have different dims. 

```python
# pseudo code for data parallelism
input_1, input_2 = torch.split(input, 2, dim=0)

output_1 = model(input_1) 
output_2 = model(input_2)

output = torch.concatenate([output_1, output_2], dim=0)  # breaks because input_1 and input_2 have different sequence lengths
```

The fix for the test is to pass the same inputs to both GPUs, and disable the stochastic duration predictor. This way, we get consistent outputs across our GPUs.

### CODE REVIEW

File: tests/models/vits/test_modeling_vits.py [hunk 1]
```diff
@@ -177,6 +179,30 @@ def test_model_forward(self):
         config_and_inputs = self.model_tester.prepare_config_and_inputs()
         self.model_tester.create_and_check_model_forward(*config_and_inputs)
 
+    @require_torch_multi_gpu
+    # override to make test deterministic across GPUs
```
Comment (ydshieh) [inline tests/models/vits/test_modeling_vits.py:183]: maybe more informative: `make all elements of the batch have the same length` (despite redundant below)

(unclear to me what `deterministic` means here)

File: tests/models/vits/test_modeling_vits.py [hunk 2]
```diff
@@ -177,6 +179,30 @@ def test_model_forward(self):
...
+    def test_multi_gpu_data_parallel_forward(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        config.use_stochastic_duration_prediction = False
+
+        # move input tensors to cuda:O
+        for key, value in inputs_dict.items():
+            if torch.is_tensor(value):
+                # make all elements of the batch the same -> ensures the output seq lengths are the same for DP
```
Comment (ydshieh) [inline tests/models/vits/test_modeling_vits.py:191]: make all elements of the batch the same **seq length**

Comment (ydshieh) [reply tests/models/vits/test_modeling_vits.py:191]: maybe mention the output length is determined by the first entry of each element in the batch.

**Review by ydshieh — APPROVED** [review]
Thanks a lot. Just a few nits for the comments.

File: tests/models/vits/test_modeling_vits.py (hunk 2 above)
Comment (sanchit-gandhi) [reply tests/models/vits/test_modeling_vits.py:191]: TWe have to make each of the elements in the batch strictly the **same** -> the output length is a function of the inputs, not just the input lengths. If the inputs are not the same, we get different output dims, so can't do DP

Comment (ydshieh) [reply tests/models/vits/test_modeling_vits.py:191]: ok, thanks for explaning!

**Review by ArthurZucker — APPROVED** [review]
Thanks! The tokenizers integration tests should rather use `k` as padding as we discussed!

File: tests/models/vits/test_modeling_vits.py (hunk 1 above)
Comment (sanchit-gandhi) [reply tests/models/vits/test_modeling_vits.py]: Resolved in https://github.com/huggingface/transformers/pull/25986/commits/63a592106f97801c1b2e1e46aa246123487e45f6

**Pull request merged**